csrc/tensor_metadata.cpp

```diff
@@ -350,6 +361,7 @@ std::vector<PolymorphicValue> GetMetaData::evaluate(
   } else {
     metadata->alloc_size = input.sizes();
     metadata->alloc_stride = input.strides();
+    // TODO: validateAllocationSizesAndStrides
```

Comment:
I don't want to do `validateAllocationSizesAndStrides` yet because this needs a `TensorDomain::noReductions(tv->getMaybeRFactorDomain());`. This is a hot path, and I want to avoid this data copying.

Reply:
In the future, I will make `TensorDomain::noReductions` non-copying by back porting C++20 ranges into C++17.